#!/usr/bin/env python

PACKAGE = "visualize_euler"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

convention_enum = gen.enum([ gen.const("SXYZ", str_t, "sxyz", "Convention"),
                       gen.const("RXYZ", str_t, "rxyz", "Convention"),
                       gen.const("SZYX_ROS_default", str_t, "szyx", "Convention"),
                       gen.const("RZYX", str_t, "rzyx", "Convention")],
                     "An enum to select euler convention")
gen.add("convention", str_t, 0, "A camera name which is edited via an enum", "sxyz", edit_method = convention_enum)

gen.add("rot_1", double_t, 0, "first dof", 0, -180, 180)
gen.add("rot_2", double_t, 0, "second dof", 0, -180, 180)
gen.add("rot_3", double_t, 0, "third dof", 0, -180, 180)

exit(gen.generate(PACKAGE, "visualize_euler", "EulerAngles"))
